File: BackEnd/src/main/java/com/tienda_ropa/ecommerce/model/Domicilio.java
package com.tienda_ropa.ecommerce.model;

import jakarta.persistence.*;
import lombok.*;
import java.util.*;

@Entity
@Table(name = "domicilio")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Domicilio extends Master {

    private String calle;
    private String numero;
    private Integer codigoPostal;
    private String referencia;

    @ManyToOne  // REMOVER cascade = CascadeType.ALL
    @JoinColumn(name = "localidad_id")
    private Localidad localidad;
}

File: FrontEnd/src/models/Domicilio.ts
import Localidad from "./Localidad";

export default class Domicilio {
    id?: number = 0;
    calle: string = "";
    numero: number = 0;
    codigoPostal: string = "";
    localidad?: Localidad;
    referencia?: string;
    eliminado!: boolean;
}